### DISCUSSION TITLE
Retrieving headers/cookies from `$http.send` response.

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#3310] [Q&A] [answered]
Hi there,
i'm trying to retrieve headers and cookies from a response returned by `$http.send`.

I've tested a naive way and added two fields to the `sendResult` struct in the Go code, `Headers map[string][]string` and `Cookies map[string]string`. Then a copied the values from the `*http.Response` object and on the Go side it looks ok.
Sadly on the JavaScript side the data isn't available, just returning numbers as keys and undefined value.

I'm not really an expert in Go so its mostly an error on my side.
Is there any way to retrieve the headers and/or cookies from the `$http.send` responses?

Greetings

**Comment by ganigeorgiev** [comment]
> Is there any way to retrieve the headers and/or cookies from the $http.send responses?

Not out of the box at the moment. I'll consider exporting them with the next minor release sometime later this week.

**Reply by ganigeorgiev** [reply, marked as answer]
_I've added the fields in `master` (https://github.com/pocketbase/pocketbase/commit/bb0a2dd69851d6732e1de67d744ff7e179c1cc93). Note that the cookie entries are exported as the parsed [`http.Cookie`](https://pocketbase.io/jsvm/interfaces/http.Cookie.html). The change will be available with the next minor v0.18.6 release sometime later this week._

**Reply by ghost** [reply]
That sounds really great, thank you for quick response and action.
